source/extensions/common/wasm/context.cc

```diff
@@ -281,6 +281,16 @@ WasmResult Context::setTickPeriod(std::chrono::milliseconds tick_period) {
   return WasmResult::Ok;
 }
 
+void Context::onCloseTCP() {
+  if (tcp_connection_closed_) {
+    return;
+  }
+  tcp_connection_closed_ = true;
+  onDone();
```

Comment:
We should check the result of `onDone` and postpone `onLog` and `onDelete` until other extensions agree, shouldn't we, @jplevyak?

Reply:
That logic is specifically for RootContexts and is already handled elsewhere.  The stream contexts are required to return "true" from the onDone call (the SDK ensure that)..   It is expected that any long lived operations which could outlast the stream will be handled by the RootContext.

Reply:
So, in that case.. no need to handle result of onDone.. Guys, if you agree with the PR, please approve and merge it. Don't have merge rights for envoy-wasm...